src/state.rs

```diff
@@ -31,15 +31,19 @@ pub enum FrontFace {
 }
 
 /// Width of a line.
-pub type LineWidth = f32;
-#[allow(missing_docs)]
-pub type OffsetFactor = f32;
-#[allow(missing_docs)]
-pub type OffsetUnits = u32;
+/// Could be f32 if not for Hash deriving issues.
+pub type LineWidth = i32;
```

Comment:
Possible, yes, and there is always a choice of using fixed-point math (as in - specify the integer value, which is 16x times what you need). However, I wouldn't worry about it - the depth offsets are not used that much (only the shadow example does), and we can patch it up later after stuff stabilizes.